### PR TITLE
Link in equinor data when running libres tests

### LINKED ISSUE #1845: Internal tests not running
**Describe the bug**
After the integration of libres none of the internal tests are running on jenkins.

**To Reproduce**
Run the tests on main

**Expected behavior**
Internal tests are running, not skipped (for example `tests/res/enkf/test_enkf_obs.py`)

### PR DESCRIPTION
**Pull request by ManInFez**
**Issue**
Resolves #1845

### CODE REVIEW

**Comment by markusdregi** [comment]
I'm slightly confused 🤷🏻

I checked Jenkins and it runs more tests on Jenkins compared to GitHub Actions. I did check some of the tests requiring Equinor test data and they all seem to run. So far, so good ✔️

However, when comparing the Jenkins log from this PR with a different, recent PR they seem to run the same number of tests 🤔 Which is odd given the issue you are trying to resolve! Also, I cannot find the specific test the issue uses as an example...

**Review by markusdregi — CHANGES_REQUESTED** [review]
Did some more digging and found the test the issue is referring to: https://github.com/equinor/ert/blob/main/libres/tests/res/enkf/test_enkf_obs.py

This test does not seem to run on Jenkins even after this change...

**Comment by ManInFez** [comment]
It does run in the ert-pr job: 
from build https://ci.equinor.com/scout/job/komodo_testing/job/ert-pr/2932
```
tests/res/enkf/test_enkf_load_results_manually.py ..                     [ 25%]
tests/res/enkf/test_enkf_obs.py ..s....                                  [ 27%]
tests/res/enkf/test_enkf_runpath.py ..                                   [ 27%]
```
One of the tests are markes as skip, due to "Abort"

**Comment by ManInFez** [comment]
found this in the other PR build: 
```
+ ln -s /project/res-testdata/ErtTestData /tmp/jenkins-scout-f_scout_ci/workspace/ert_PR/test-data/Equinor
ln: failed to create symbolic link ‘/tmp/jenkins-scout-f_scout_ci/workspace/ert_PR/test-data/Equinor/ErtTestData’: Permission denied
```

File: libres/testjenkins.sh [hunk 1]
```diff
@@ -109,7 +108,6 @@ clone_repos () {
 	echo "Cloning into $LIBRES_ROOT"
 	git clone . $LIBRES_ROOT
 	mkdir -p $LIBRES_BUILD
-	ln -s /project/res-testdata/ErtTestData $LIBRES_ROOT/../test-data/Equinor
```
Comment (markusdregi) [inline libres/testjenkins.sh:112]: Why are we removing instead of fixing this?

Comment (ManInFez) [reply libres/testjenkins.sh:112]: I am doing some tests. Adding equinor-data was done three places. One in the cloning, one at compile time, and one right before testing. Only one of those are needed. I am suspecting that trying to do this redundantly might cause some issues. Will poke you when I find out. I suspect that the compile time is never used as LD_LIBRARY_PATH is not set to the newly built libres, and that the second linking causes the run_pytest_equinor is terminated when the linking fails.

Comment (ManInFez) [reply libres/testjenkins.sh:112]: This worked: (https://ci.equinor.com/scout/job/ert_PR/1319/console)
```
../../libres/tests/res/enkf/test_ecl_config.py ...                       [  5%]
../../libres/tests/res/enkf/test_enkf_fs.py ....                         [ 12%]
../../libres/tests/res/enkf/test_enkf_library.py ..                      [ 15%]
../../libres/tests/res/enkf/test_enkf_load_results_manually.py ..        [ 19%]
../../libres/tests/res/enkf/test_enkf_obs.py ..s....                     [ 31%]
../../libres/tests/res/enkf/test_ert_context.py ...                      [ 36%]
../../libres/tests/res/enkf/test_field_export.py s...                    [ 43%]
../../libres/tests/res/enkf/test_labscale.py ss                          [ 47%]
../../libres/tests/res/enkf/test_summary_key_set.py ....                 [ 54%]
../../libres/tests/res/enkf/test_update.py s                             [ 56%]
../../libres/tests/res/enkf/data/test_gen_data.py .                      [ 57%]
../../libres/tests/res/enkf/data/test_gen_data_config.py ..              [ 61%]
../../libres/tests/res/enkf/data/test_gen_kw_config_equinor.py .         [ 63%]
../../libres/tests/res/enkf/export/test_arg_loader.py .                  [ 64%]
../../libres/tests/res/enkf/plot/test_plot_data.py ...                   [ 70%]
../../libres/tests/res/fm/test_ecl_run.py ........                       [ 84%]
../../libres/tests/res/fm/test_ecl_run_new.py .........                  [100%]
```
but I will test some more

**Comment by ManInFez** [comment]
did it the right way, by only using the compile option and installing libecl/libres into the virtual environment

**Comment by markusdregi** [comment]
Nice! Have you verified that the internal tests are now running for both of the Jenkins jobs?

**Comment by ManInFez** [comment]
ert_PR job runs only libres tests:
```
ctests:
100% tests passed, 0 tests failed out of 146
pytest_normal:
==== 380 passed, 9 skipped, 57 deselected, 8 warnings in 256.54s (0:04:16) =====
pytest_equinor:
==== 52 passed, 5 skipped, 389 deselected, 8 warnings in 120.38s (0:02:00) =====
```

ert_pr job runs both libres and ert tests:
```
ert:
= 716 passed, 1 skipped, 5 deselected, 2 xfailed, 87 warnings in 565.88s (0:09:25) =
libres:
=========== 411 passed, 14 skipped, 7 warnings in 477.65s (0:07:57) ============
```
Some of these things do not add up :shrug:

**Comment by markusdregi** [comment]
I used the following two tests as indicators to verify that both the C and Python tests run with the internal data for both of the Jenkins jobs: `test_enkf_obs` (Python) and `rms_file_test` (C).

Neither of the two runs during GitHub Actions, which supports my theory after inspecting the tests that these are indeed requiring Equinor data ✅ 

The ERT non-komodo build sports both of the two tests ✅ 

However, the Komodo build does only run `test_enkf_obs` ✅, but not `rms_file_test` 🛑. It also does not seem like that job is symlinking in the test data. I would prefer if you could change that...

**Review by markusdregi — APPROVED** [review]
After a chat with Åsmund I see that this PR makes both the Jenkins jobs run internal tests for the tests that the jobs are running. A separate issue will be made on investigating whether we can run the c tests against komodo and to setup the vanilla Jenkins build to also run the ERT tests.

**Pull request merged**